gbn/gbn_conn.go

```diff
@@ -251,21 +241,21 @@ func (g *GoBackNConn) start() {
 	g.log.Debugf("Starting")
 
 	pingTime := time.Duration(math.MaxInt64)
-	if g.cfg.pingTime != 0 {
-		pingTime = g.cfg.pingTime
+	if g.timeoutManager.GetPingTime() != 0 {
+		pingTime = g.timeoutManager.GetPingTime()
 	}
 
 	g.pingTicker = NewIntervalAwareForceTicker(pingTime)
 	g.pingTicker.Resume()
 
 	pongTime := time.Duration(math.MaxInt64)
-	if g.cfg.pongTime != 0 {
-		pongTime = g.cfg.pongTime
+	if g.timeoutManager.GetPongTime() != 0 {
+		pongTime = g.timeoutManager.GetPongTime()
 	}
```

Comment:
move all this logic into the timeout manager's GetPongTime msg?

Reply:
same with pingTime above

Reply:
True thanks, it's been addressed by the latest push.